src/NuGet.Clients/PackageManagement.UI/Feeds/MultiSourcePackageMetadataProvider.cs

```diff
@@ -204,8 +204,21 @@ internal class MultiSourcePackageMetadataProvider : IPackageMetadataProvider
                 .Select(task => task.ContinueWith(LogError, TaskContinuationOptions.OnlyOnFaulted))
                 .ToArray();
 
-            var completed = (await Task.WhenAll(tasks))
-                .Where(m => m != null);
+            IEnumerable<IPackageSearchMetadata> completed = null;
+            try
+            {
+                completed = (await Task.WhenAll(tasks))
+                    .Where(m => m != null);
+            }
+            catch
+            {
+                completed = tasks.Where(m => m != null && !m.IsFaulted && m.Result != null).Select(m => m.Result);
```

Comment:
`m` is a `Task` so no need to check for null.
Check if it `IsCompleted` and not `IsFaulted` and not `IsCancelled`.